spec.go

```diff
@@ -204,16 +204,237 @@ type SymlinkTarget struct {
 	Group string `yaml:"group,omitempty" json:"group,omitempty"`
 }
 
+// GomodEdits groups the go.mod manipulation directives that can be applied
+// before downloading module dependencies.
+type GomodEdits struct {
+	// Replace applies go.mod replace directives before downloading module dependencies.
+	// Each entry can be either a string "old:new" or a struct with Old and New fields.
+	Replace []GomodReplace `yaml:"replace,omitempty" json:"replace,omitempty"`
+	// Require applies go.mod require directives before downloading module dependencies.
+	// Each entry can be either a string "module:version" or a struct with Module and Version fields.
+	Require []GomodRequire `yaml:"require,omitempty" json:"require,omitempty"`
+}
+
+// GomodReplace represents a go.mod replace directive.
+// It can be specified as a string "old:new" or as a struct with Old and New fields.
+// This allows users to substitute module dependencies before go mod download runs,
+// useful for pointing to local forks or alternate versions.
+type GomodReplace struct {
+	// Old is the module path to replace (can include @version)
+	Old string `yaml:"old" json:"old"`
+	// New is the replacement module path or local directory
+	New string `yaml:"new" json:"new"`
+}
+
+func (r GomodReplace) String() string {
+	return r.Old + ":" + r.New
+}
+
+func (r GomodReplace) goModEditArg() (string, error) {
+	if r.Old == "" || r.New == "" {
+		return "", errors.Errorf("invalid gomod replace, old and new must be non-empty")
+	}
+	return r.Old + "=" + r.New, nil
+}
+
+func (r *GomodReplace) UnmarshalYAML(b []byte) error {
```

Comment:
In these we'll need to also set source maps, which would also require using the ast version of UnmarshalYAML (see other UnmarshalYAML's in the repo).